streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -440,16 +440,16 @@ private void close(final boolean clean) {
             }
 
             if (state() == State.CLOSING) {
-                // first close state manager (which is idempotent) then close the record collector (which could throw),
-                // if the latter throws and we re-close dirty which would close the state manager again.
-                StateManagerUtil.closeStateManager(log, logPrefix, clean, stateMgr, stateDirectory);
-
                 // if EOS is enabled, we wipe out the whole state store for unclean close
                 // since they are invalid to use anymore
                 if (!clean && !eosDisabled) {
                     StateManagerUtil.wipeStateStores(log, stateMgr);
                 }
 
+                // first close state manager (which is idempotent) then close the record collector (which could throw),
+                // if the latter throws and we re-close dirty which would close the state manager again.
+                StateManagerUtil.closeStateManager(log, logPrefix, clean, stateMgr, stateDirectory);
```

Comment:
Is that okay that we wipe out the directory already and then closing state manager, in which we would flush the stores and then close them? Would any exception be thrown?